src/Components/Search/NewSearch/NewSearchBarInput.tsx

```diff
@@ -66,6 +71,25 @@ const NewSearchBarInput: FC<NewSearchBarInputProps> = ({
     )
   }
 
+  const handlePillClick = (pill: PillType) => {
+    setSelectedPill(pill)
+
+    relay.refetch(
```

Comment:
**suggestion**: I think we can combine all refetch calls into one (feel free to ignore if you don't agree with the suggestions). Like this

```javascript
const refetch = (term: string, entity: SearchEntity) => {
  relay.refetch(
    {
      hasTerm: true,
      term,
      entities: [entity],
    },
    null,
    error => {
      if (error) {
        logger.error(error)
        return
      }
    }
  )
}

const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
  const term = event.target.value
  const entity = ELASTIC_PILL_KEY_TO_SEARCH_ENTITY?.[selectedPill.key]
  
  setValue(term)
  refetch(term, entity)
}

const handlePillClick = (pill: PillType) => {
  const entity = ELASTIC_PILL_KEY_TO_SEARCH_ENTITY?.[pill.key]
  
  setSelectedPill(pill)
  refetch(value, entity)
}
```

or even better like this
```javascript
useEffect(() => {
  const entity = ELASTIC_PILL_KEY_TO_SEARCH_ENTITY?.[selectedPill.key]
  
  relay.refetch(
    {
      hasTerm: true,
      term: value,
      entities: [entity],
    },
    null,
    error => {
      if (error) {
        logger.error(error)
        return
      }
    }
  )
}, [value, selectedPill])

const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
  setValue(event.target.value)
}

const handlePillClick = (pill: PillType) => {
  setSelectedPill(pill)
}
```